packages/vite/src/node/server/index.ts

```diff
@@ -556,9 +546,6 @@ export async function _createServer(
 
   watcher.on('change', async (file) => {
     file = normalizePath(file)
-    if (file.endsWith('/package.json')) {
-      return invalidatePackageData(packageCache, file)
-    }
```

Comment:
This return call was blocking HMR updates for `package.json`, that also blocks `handleHotUpdate`, so removed here.